com.unity.render-pipelines.universal/Runtime/TestCameraProperties.cs

```diff
@@ -0,0 +1,165 @@
+using UnityEngine;
+using Unity.Mathematics;
+using UnityEngine.Rendering;
+
+// TODO: Remove after the testing
+internal static class TestCameraProperties
+{
+    private static Mesh SomeMesh;
+    private static Material FetchCameraPropertiesMaterialOld;
+    private static Material FetchCameraPropertiesMaterialNew;
+    private static ComputeBuffer CameraPropertiesDataBufferOld;
+    private static ComputeBuffer CameraPropertiesDataBufferNew;
+    private static CameraPropertiesData DataOld;
+    private static CameraPropertiesData DataNew;
+
+    public struct CameraPropertiesData
+    {
+        public float4 _WorldSpaceCameraPos;
+        public float4x4 _Reflection;
+        public float4 _ScreenParams;
+        public float4 _ProjectionParams;
+        public float4 _ZBufferParams;
+        public float4 unity_OrthoParams;
+        public float4 unity_HalfStereoSeparation;
+        public float4 unity_CameraWorldClipPlanes0;
+        public float4 unity_CameraWorldClipPlanes1;
+        public float4 unity_CameraWorldClipPlanes2;
+        public float4 unity_CameraWorldClipPlanes3;
+        public float4 unity_CameraWorldClipPlanes4;
+        public float4 unity_CameraWorldClipPlanes5;
+        public float4 unity_BillboardNormal;
+        public float4 unity_BillboardTangent;
+        public float4 unity_BillboardCameraParams;
+    }
+
+    public static void SampleCameraPropertiesOld(CommandBuffer cmd)
+    {
+        if (SomeMesh == null)
+            SomeMesh = CoreUtils.CreateCubeMesh(-Vector3.one, Vector3.one);
+
+        unsafe
+        {
+            if (FetchCameraPropertiesMaterialOld == null)
+            {
+                FetchCameraPropertiesMaterialOld = new Material(Shader.Find("Hidden/FetchCameraPropertiesShader"));
+                CameraPropertiesDataBufferOld = new ComputeBuffer(1, sizeof(CameraPropertiesData));
+                cmd.SetRandomWriteTarget(1, CameraPropertiesDataBufferOld);
+            }
+        }
+        cmd.DrawMesh(SomeMesh, Matrix4x4.Translate(Vector3.zero), FetchCameraPropertiesMaterialOld);
+
+        CameraPropertiesData[] array = new CameraPropertiesData[1];
+        CameraPropertiesDataBufferOld.GetData(array);
+        DataOld = array[0];
+    }
+
+    public static void SampleCameraPropertiesNew(CommandBuffer cmd)
```

Comment:
Can't we merge these two functions with some arguments? Makes it easier to maintain imo.

Reply:
Yeap, but this is test script, which currently wont be landed to master.